hive/hive-2.2.0/src/main/java/io/tidb/bigdata/hive/TiDBInputSplit.java

```diff
@@ -87,46 +64,38 @@ public long getLength() {
   @Override
   public void write(DataOutput dataOutput) throws IOException {
     dataOutput.writeUTF(path.toString());
-    dataOutput.writeUTF(databaseName);
-    dataOutput.writeUTF(tableName);
-    dataOutput.writeUTF(startKey);
-    dataOutput.writeUTF(endKey);
-    dataOutput.writeLong(physicalTimestamp);
-    dataOutput.writeLong(logicalTimestamp);
+    dataOutput.writeInt(splitInternals.size());
+    for (int i = 0; i < splitInternals.size(); i++) {
+      dataOutput.writeUTF(splitInternals.get(i).getTable().getSchemaName());
+      dataOutput.writeUTF(splitInternals.get(i).getTable().getTableName());
+      dataOutput.writeUTF(splitInternals.get(i).getStartKey());
+      dataOutput.writeUTF(splitInternals.get(i).getEndKey());
+      dataOutput.writeLong(splitInternals.get(i).getTimestamp().getPhysical());
+      dataOutput.writeLong(splitInternals.get(i).getTimestamp().getLogical());
+    }
   }
 
   @Override
   public void readFields(DataInput dataInput) throws IOException {
     this.path = new Path(dataInput.readUTF());
-    this.databaseName = dataInput.readUTF();
-    this.tableName = dataInput.readUTF();
-    this.startKey = dataInput.readUTF();
-    this.endKey = dataInput.readUTF();
-    this.physicalTimestamp = dataInput.readLong();
-    this.logicalTimestamp = dataInput.readLong();
-  }
-
-  public String getDatabaseName() {
-    return databaseName;
-  }
-
-  public String getTableName() {
-    return tableName;
+    Integer size = dataInput.readInt();
+    List<SplitInternal> splitInternalList = new ArrayList<>();
```

Comment:
It would be better to use `new ArrayList<>(size)` to avoid array capacity expansion for big table.

Reply:
I have fixed it, please review again.